util/mp.py

```diff
@@ -1,10 +1,51 @@
 from __future__ import absolute_import, division, print_function
 
+import os
+
 import future.moves.itertools as itertools
+import psutil
 
 import libtbx.easy_mp
 
 
+def available_nproc():
+    """
+    Choose a suitable value for nproc based on the availability of processors.
+    """
+
+    nproc = os.environ.get("NSLOTS", 0)
+    try:
+        nproc = int(nproc)
+    except ValueError:
+        nproc = 0
+    if nproc >= 1:
+        return nproc
```

Comment:
```suggestion
        nproc = int(nproc)
        if nproc >= 1:
            return nproc
    except ValueError:
        pass
```

Python 3 does not allow string / int comparison, so only consider returning the value in the `try` block